solidity/src/proof_plans/SortMergeJoinExec.presl

```diff
@@ -0,0 +1,418 @@
+                    )
+
+                // Finally, we can do our membership check
+                pop(
+                    membership_check_evaluate(
+                        builder_ptr,
+                        mload(alpha_and_beta),
+                        mload(add(alpha_and_beta, WORD_SIZE)),
+                        chi_eval,
+                        res_chi_eval,
+                        hat_evals,
+                        right_column_evals
+                    )
+                )
+            }
+            function evaluate_consume_and_check_left_column_evals(plan_ptr, builder_ptr, alpha_and_beta, res_chi_eval, left_right_and_output_evaluations, chi_eval) -> plan_ptr_out {
+                // We only need the hat evals for a membership check on the output columns
+                let hat_evals, join_evals
+                plan_ptr_out, hat_evals, join_evals :=
+                    evaluate_input_plans(plan_ptr, builder_ptr, mload(left_right_and_output_evaluations))
+                mstore(left_right_and_output_evaluations, join_evals)
+                let res_column_evals :=
+                    consume_and_membership_check_left_column_evals(
+                        builder_ptr, alpha_and_beta, hat_evals, res_chi_eval, chi_eval
+                    )
+                mstore(add(left_right_and_output_evaluations, WORDX2_SIZE), res_column_evals)
+            }
+            function evaluate_consume_and_check_right_join_evals(plan_ptr, builder_ptr, alpha_and_beta, res_chi_eval, left_right_and_output_evaluations, chi_eval) -> plan_ptr_out {
+                let hat_evals, i_eval, join_evals
+                plan_ptr, hat_evals, join_evals :=
+                    evaluate_input_plans(plan_ptr, builder_ptr, mload(add(left_right_and_output_evaluations, WORD_SIZE)))
+                i_eval :=
+                    consume_and_membership_check_right_column_evals(
+                        builder_ptr,
+                        alpha_and_beta,
+                        mload(join_evals),
+                        hat_evals,
+                        res_chi_eval,
+                        chi_eval,
+                        left_right_and_output_evaluations
+                    )
+                mstore(add(left_right_and_output_evaluations, WORD_SIZE), join_evals)
+
+                monotonic_verify(
+                    builder_ptr,
+                    mload(alpha_and_beta),
+                    mload(add(alpha_and_beta, WORD_SIZE)),
+                    i_eval,
+                    res_chi_eval,
+                    1,
+                    1
+                )
+                plan_ptr_out := plan_ptr
+            }
+
+            function evaluate_remaining_check_on_both_left_and_right_sides(plan_ptr, builder_ptr, alpha_and_beta, left_right_and_output_evaluations, left_chi_eval, right_chi_eval) -> plan_ptr_out, w_eval {
```

Comment:
Let $m$ be the number of rows of the join result. Check that `u` is strictly increasing, elements of `left_join_columns` and `right_join_columns` are both elements of `u` and that the multiplicities, `w_l` and `w_r` for these membership checks, satisfy the constraint `sum w_l * w_r - chi_m = 0`